### PR TITLE
:seedling: Add API types for DiscoverVariables hook

### PR DESCRIPTION
**Pull request by killianmuldoon**
Add the API types proposed for the DiscoverVariables extension to the runtime API package.

Part of #7985

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
/area runtime-sdk

File: exp/runtime/hooks/api/v1alpha1/topologymutation_types.go [hunk 1]
```diff
@@ -179,6 +180,32 @@ type HolderReference struct {
...
+	metav1.TypeMeta `json:",inline"`
+
+	// CommonResponse contains Status and Message fields common to all response types.
+	CommonResponse `json:",inline"`
+
+	// Variables are variable schemas for variables defined by the DiscoverVariables hook.
+	Variables []clusterv1.ClusterClassVariable `json:"variables"`
+}
+
+var _ ResponseObject = &DiscoveryResponse{}
```
Comment (ykakarap) [inline exp/runtime/hooks/api/v1alpha1/topologymutation_types.go:204]:
```suggestion
var _ ResponseObject = &DiscoverVariablesResponse{}
```

**Review by ykakarap — COMMENTED** [review]
Overall looks good.

File: exp/runtime/hooks/api/v1alpha1/topologymutation_types.go [hunk 2]
```diff
@@ -22,6 +22,7 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
```
[the remark this answers is not in the document]
Comment (sbueringer) [reply exp/runtime/hooks/api/v1alpha1/topologymutation_types.go:25]: Thx for bringing it up, I think same as for lifecycle hooks it's fine to have this dependency for now. Means we have to evolve them together but that should be okay for now.

**Comment by sbueringer** [comment]
Thx!

/lgtm
/approve

**Pull request merged**